### ISSUE TITLE
UTF-8: Implement quoting and escaping in UI autocompletion

### ISSUE THREAD

**Issue by ywwg** [prometheus/prometheus#15236]
### Proposal

as part of https://github.com/prometheus/prometheus/issues/15202, we will need to update the autocomplete engine in the UI to know when escaping or quoting is necessary. @juliusv felt this was a big enough subtask to call out separately.

**Comment by ywwg** [comment]
not blocking for v3

**Comment by ywwg** [comment]
@anshdeep0504 yes this issue is still open. I do not know anything about the implementation but there is a main design question:

how should the UI behave when autocompleting? Once a metric name requires quoting, should the UI pre-fill the `{"` in front of the string, or should it just add those after the autocomplete is selected? or, should the UI just decorate the accepted completion with `{"` once the completion is selected?  It depends what is a nicer user experience.

**Comment by ywwg** [comment]
@anshdeep0504 any update on this?

**Comment by sujalshah-bit** [comment]
can  I work on this ?

**Comment by sujalshah-bit** [comment]
Every time I make any changes in `HistoryCompleteStrategy` this error keep comming
```console
[LRU_CACHE_UNBOUNDED] UnboundedCacheWarning: TTL caching without ttlAutopurge, max, or maxSize can result in unbounded memory consumption. Error Component Stack
```
related #15396

**Comment by bboreham** [comment]
Hello from the bug-scrub!

@juliusv can you offer any help to @sujalshah-bit ?

**Comment by Nexucis** [comment]
@sujalshah-bit can you explain how you are changing / updating the function `HistoryCompleteStrategy` ? There is no such error or warning in the Prometheus UI currently. So it should be something you are doing or not setting in the config.

Please provide more context so we can provide you meaningful advices.
